pkg/servicemesh/federation/server/server.go

```diff
@@ -486,6 +508,9 @@ func (s *meshServer) resync() {
 			s.addService(svcKey, svcMessage)
 		}
 	}
+	if err := s.statusHandler.Flush(); err != nil {
```

Comment:
do we need to manually `Flush()` from outside the status handler? couldn't this be a private func as well?

Reply:
nevermind I see now that we flush manually here, I guess because there might be >1 event at a time. I think I'd prefer the flush() method to be private and triggering it automatically (using debounce) from within statushandler - but that's an optimization

Reply:
I chose to have the status automatically written for important events, like connect/disconnect and full sync, and to let the periodic updates write out status for other events.  The main reason I did this was to avoid spamming the api server when processing a full sync.  This means the user will immediately be aware of the connection status, as well as the imported/exported services after full syncs.  Interim events will not be immediately reflected in the status (e.g. a new service is created in the mesh and gets exported).